### PR TITLE
Fetching playlist from API in InputPage.tsx and rendering in PlaylistPage.tsx

### PR DESCRIPTION
**Pull request by maxitect**
* Update fetchPlaylist function with types and returns
* Initialise context variables for playlist json within Content.tsx
* Write to context variables for playlist json within InputPage.tsx
* Read to context variables for playlist json within PlaylistPage.tsx

### CODE REVIEW

File: src/pages/InputPage.tsx [hunk 1]
```diff
@@ -94,13 +98,19 @@ function InputPage({ onNext }: InputProps) {
 								}
 							/>
 						</div>
-						{error && <p className="error">{error}</p>}
```
Comment (maxitect) [inline src/pages/InputPage.tsx:97]: @JasonWarrenUK I implemented all your changes apart from this line I couldn't see how it fits in with the rest, feel free to make a suggestion that I can commit

File: src/pages/InputPage.tsx [hunk 2]
```diff
@@ -50,24 +46,32 @@ function InputPage({ onNext }: InputProps) {
...
-	const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+  
+	const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
 		event.preventDefault();
 
-		const formData = {
+		// Trigger loading page
+		setLoading(true);
+
+		// Update user response state based on form input
```
Comment (JasonWarrenUK) [inline src/pages/InputPage.tsx:56]:
```suggestion
		setLoading(true);
```

File: src/pages/InputPage.tsx (hunk 1 above)
Comment (JasonWarrenUK) [reply src/pages/InputPage.tsx:97]: So this line basically means:

if (there is an error message) {
	show this p element
}

Comment (JasonWarrenUK) [reply src/pages/InputPage.tsx:97]: So I think you can keep it as is

Comment (JasonWarrenUK) [reply src/pages/InputPage.tsx:97]:
```suggestion
						{error && <p className="error">{error}</p>}
```

**Review by JasonWarrenUK — APPROVED** [review]

**Pull request merged**